src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity) {
+        Storage.fruits.put(fruit, Storage.fruits.get(fruit) - quantity);
```

Comment:
Consider adding a check to ensure that the quantity to be purchased does not exceed the available stock. This will prevent the storage from having negative quantities, which is likely an unintended behavior.